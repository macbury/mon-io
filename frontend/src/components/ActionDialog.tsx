import React from 'react'
import { useTranslation } from 'react-i18next'
import { Dialog, Portal, List } from 'react-native-paper'
import styled from 'styled-components/native'

const Container = styled(Dialog)`
  max-width: 400px;
  min-width: 300px;
  display: flex;
  align-self: center;
  flex-shrink: 1;
  margin-top: 20px;
  margin-bottom: 20px;
`

const InnerContainer = styled.View`
  border-top-width: 1px;
  border-top-color: ${({ theme }) => theme.headerBorderColor};
`

const Icon = styled(List.Icon)`
`

export interface IAction {
  title: string
  subtitle?: string
  icon: string
  hidden?: boolean
  onPress()
}

export interface IActionDialogProps {
  visible: boolean
  title: string
  items: IAction[]
  onDismiss()
}

export default function ActionDialog({ visible, onDismiss, title, items } : IActionDialogProps) {
  const { t } = useTranslation()

  const list = items.filter(({ hidden }) => !hidden).map(({ title, subtitle, icon, onPress }) => {
    return (
      <List.Item
        key={title}
        title={title}
        description={subtitle}
        onPress={onPress}
        left={props => <Icon icon={icon} />} />
    )
  })

  return (
    <Portal>
      <Container visible={visible} onDismiss={onDismiss} dismissable>
        <Dialog.Title>{title}</Dialog.Title>
        <InnerContainer>
          {list}
          <List.Item
            title={t('dialogs.budget_actions.actions.done')}
            onPress={onDismiss}
            left={props => <Icon icon="check-circle" />} />
        </InnerContainer>
      </Container>
    </Portal>
  )
}