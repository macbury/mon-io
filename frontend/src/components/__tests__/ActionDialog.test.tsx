import React from 'react'
import { shallow } from 'enzyme'
import snapshot from 'enzyme-to-json'
import ActionDialog, { IActionDialogProps } from '../ActionDialog'

it('renders correctly', () => {
  const props : IActionDialogProps = {
    visible: true,
    title: 'Dialog title',
    onDismiss: jest.fn(),
    items: [
      {
        title: 'First item',
        icon: 'repeat',
        onPress: jest.fn(),
        subtitle: 'Subtitle'
      },

      {
        title: 'Second item',
        icon: 'repeat',
        onPress: jest.fn(),
        subtitle: 'Subtitle'
      }
    ]
  }

  const component = shallow(<ActionDialog {...props}/>)
  expect(snapshot(component)).toMatchSnapshot()
})