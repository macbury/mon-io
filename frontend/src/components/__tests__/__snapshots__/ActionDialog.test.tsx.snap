// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<withTheme(Portal)>
  <Styled(withTheme(Dialog))
    dismissable={true}
    onDismiss={[MockFunction]}
    visible={true}
  >
    <withTheme(Dialog.Title)>
      Dialog title
    </withTheme(Dialog.Title)>
    <Styled(View)>
      <withTheme(List.Item)
        description="Subtitle"
        key="First item"
        left={[Function]}
        onPress={[MockFunction]}
        title="First item"
      />
      <withTheme(List.Item)
        description="Subtitle"
        key="Second item"
        left={[Function]}
        onPress={[MockFunction]}
        title="Second item"
      />
      <withTheme(List.Item)
        left={[Function]}
        onPress={[MockFunction]}
        title="Done"
      />
    </Styled(View)>
  </Styled(withTheme(Dialog))>
</withTheme(Portal)>
`;
